### PR TITLE
[dv,chip_sw] Add clkmgr_escalation_reset test

### PR DESCRIPTION
**Pull request by matutem**
Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM
@cindychip can also take a look at the alert_handler portion

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+/**
+ * Program the alert handler to escalate on alerts upto phase 2 (i.e. reset).
+ * Also program the aon timer with:
+ * - bark after escalation starts, so the interrupt is suppressed by escalation,
+ * - bite after escalation reset, so we should not get timer reset.
+ */
+enum {
+  kWdogBarkMicros = 3 * 100,          // 200 us
+  kWdogBiteMicros = 8 * 100,          // 400 us
```
Comment (cindychip) [inline sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:39]: nit: comment should be 800 us right?

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = udiv64_slow(
+           kEscalationPhase0Micros * kClockFreqPeripheralHz, 1000000, NULL)},
+      {.phase = kDifAlertHandlerClassStatePhase1,
+       .signal = 1,
+       .duration_cycles = udiv64_slow(
+           kEscalationPhase1Micros * kClockFreqPeripheralHz, 1000000, NULL)},
+      {.phase = kDifAlertHandlerClassStatePhase2,
+       .signal = 3,
```
Comment (cindychip) [inline sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:163]: I guess it is intentional to use signal 3 at phase 2 here? Just want to double check ;)

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+  if (rst_info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Booting for the first time, starting test");
+    execute_test(&aon_timer);
+  } else if (rst_info == kDifRstmgrResetInfoEscalation) {
+    LOG_INFO("Booting for the second time due to escalation reset");
+    // Check the clkmgr has no fatal errors due to reset.
+    dif_clkmgr_fatal_err_codes_t codes;
+    CHECK_DIF_OK(dif_clkmgr_fatal_err_code_get_codes(&clkmgr, &codes));
+    CHECK(codes == 0, "Fatal error codes should be cleared upon reset");
```
Comment (cindychip) [inline sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:256]: In testplan, I saw it mentioned:
```Upon alert escalation reset, the internal status should be clear and clkmgr should not attempt to send out more alerts.
```
I wonder if we can read the alert_cause register and make sure the value is 0?

**Review by cindychip — COMMENTED** [review]
LG with a few minor suggestions

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c (hunk 1 above)
Comment (matutem) [reply sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c]: Fixed, thanks.

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c (hunk 2 above)
Comment (matutem) [reply sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:163]: It is intentional.

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  if (peripheral == kTopEarlgreyPlicPeripheralAonTimerAon) {
+    uint32_t irq =
+        (irq_id - (dif_rv_plic_irq_id_t)
+                      kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+    // We should not get aon timer interrupts since escalation suppresses them.
+    LOG_ERROR("Unexpected aon timer interrupt %0d", irq);
```
Comment (moidx) [inline sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:90]: nit: Should this be a CHECK evaluating to false to abort the test case?

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+ */
+static void alert_handler_config(void) {
+  dif_alert_handler_alert_t alerts[] = {kTopEarlgreyAlertIdClkmgrAonFatalFault};
+  dif_alert_handler_class_t alert_classes[] = {kDifAlertHandlerClassA};
+
+  dif_alert_handler_escalation_phase_t esc_phases[] = {
+      {.phase = kDifAlertHandlerClassStatePhase0,
+       .signal = 0,
+       .duration_cycles = udiv64_slow(
+           kEscalationPhase0Micros * kClockFreqPeripheralHz, 1000000, NULL)},
```
Comment (moidx) [inline sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:157]: nit: consider the recommendations from the style guide regarding argument comments: https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

e.g. 

```c++
udiv64_slow(
           kEscalationPhase0Micros * kClockFreqPeripheralHz, 
           kClockFreqDivisor, /*rem_out=*/NULL);
```

**Review by moidx — APPROVED** [review]

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c (hunk 3 above)
Comment (matutem) [reply sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:256]: Good catch! Done.

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c (hunk 5 above)
Comment (matutem) [reply sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:158]: Good call. Done here and in a couple more files.

File: sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c (hunk 4 above)
Comment (matutem) [reply sw/device/tests/sim_dv/clkmgr_escalation_reset_test.c:92]: Done.

**Pull request merged**